File: frontend/console/src/api/metadata.js
import Axios from "@/http";

export function createMetadata(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/metadata', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveMetadata(metadataId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/metadata/${ metadataId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateMetadata(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/metadata', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteMetadata(metadataId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/metadata/${ metadataId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteMetadataWithFlag(metadataId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/metadata/${ metadataId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllMetadatas() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/metadata/all').then(res => {
            resolve(res)
        })
    })
}

export function queryMetadata (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/metadata/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/metadata.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_METADATA = "/config/metadata"
const URL_ROUTER_METADATA_BY_ID = "/config/metadata/:metadata-id"
const URL_ROUTER_METADATA_QUERY = "/config/metadata/query"
const URL_ROUTER_METADATA_ALL = "/config/metadata/all"

func MetadataRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_METADATA, config.CreateMetadata)
	r.GET(URL_ROUTER_METADATA_BY_ID, config.RetrieveMetadata)
	r.PUT(URL_ROUTER_METADATA, config.UpdateMetadata)
	r.DELETE(URL_ROUTER_METADATA_BY_ID, config.DeleteMetadata)
	r.PUT(URL_ROUTER_METADATA_BY_ID, config.DeleteMetadataWithFlag)
	r.POST(URL_ROUTER_METADATA_QUERY, config.QueryMetadata)
	r.GET(URL_ROUTER_METADATA_ALL, config.GetAllMetadatas)
	return r
}